src/features/embalm/stepContent/hooks/useCreateSarcohpagus.ts

```diff
@@ -31,103 +52,183 @@ export function useCreateSarcophagus() {
     try {
-      // Prepare the payload
-      if (!canCreateSarcophagus) return emptyData;
-
       // Step 1: Split the outer layer private key using shamirs secret sharing
       const shards: Uint8Array[] = split(outerPrivateKey, {
         shares: selectedArchaeologists.length,
-        threshold: requiredArchaeologists,
+        threshold: requiredArchaeologists
       });
 
       // Step 2: Encrypt each shard of the outer layer private key using each archaeologist's public
       // key
       const archPublicKeys = selectedArchaeologists.map(x => x.publicKey!);
-      const encryptedShards = await encryptShards(archPublicKeys, shards);
+      const encShards = await encryptShards(archPublicKeys, shards);
 
       // Step 3: Upload the encrypted shards mapping to the arweave bundlr
-      const mapping: Record<string, string> = encryptedShards.reduce(
+      const mapping: Record<string, string> = encShards.reduce(
         (acc, shard) => ({
           ...acc,
-          [shard.publicKey]: shard.encryptedShard,
+          [shard.publicKey]: shard.encryptedShard
         }),
         {}
       );
 
-      const encryptedShardsTxId = await uploadArweaveFile(Buffer.from(JSON.stringify(mapping))); // TODO: change to use uploadFile for Bundlr, once local testing figured out
+      const txId = await uploadArweaveFile(Buffer.from(JSON.stringify(mapping))); // TODO: change to use uploadFile for Bundlr, once local testing figured out
 
-      dispatch(setShardPayloadData(encryptedShards, encryptedShardsTxId));
-      return { encryptedShards, encryptedShardsTxId };
+      setArchaeologistShards(encShards);
+      setEncryptedShardsTxId(txId);
     } catch (error) {
       console.error(error);
-      return emptyData;
-    } finally {
-      dispatch(setIsUploading(false));
     }
   }, [
-    canCreateSarcophagus,
     requiredArchaeologists,
     outerPrivateKey,
     selectedArchaeologists,
     uploadArweaveFile,
+    dispatch
+  ]);
+
+  const uploadAndSetDoubleEncryptedFile = useCallback(async () => {
+    const payload = await readFileDataAsBase64(file!);
+
+    // Step 1: Encrypt the inner layer
+    const encryptedInnerLayer = await encrypt(recipientState.publicKey, payload); // TODO: Restore this line and remove line below when in-app create sarco flow is ready
+
+    // Step 2: Encrypt the outer layer
+    const encryptedOuterLayer = await encrypt(outerPublicKey!, encryptedInnerLayer);
+
+    // Step 3: Upload the double encrypted payload to the arweave bundlr
+    const payloadTxId = await uploadArweaveFile(encryptedOuterLayer); // TODO: change to use uploadFile for Bundlr, once local testing figured out
+
+    setSarcophagusPayloadTxId(payloadTxId);
+  }, [
     dispatch,
+    file,
+    outerPublicKey,
+    recipientState.publicKey,
+    uploadArweaveFile,
+    setSarcophagusPayloadTxId
   ]);
 
-  const handleCreate = useCallback(async () => {
-    // Step 8: Create the sarcophagus
+  // TODO -- add approval step
```

Comment:
the switch statement in the `useEffect` below is not the most `React`y way to do this, but it does work correctly (each steps run in order and only once).